lib/webtask.js

```diff
@@ -708,6 +748,48 @@ exports.ensure_authenticated = function(ctx) {
             });
         }
 
+        function validate_auth_token(auth) {
+            var auth_token;
+            var decoded_auth_token;
+            var userinfo;
+            var secret = new Buffer((auth.secret || DEFAULT_AUTH0_CLIENT_SECRET), 'base64');
+
+            try {
+                auth_token = req.query.key || (req.headers['authorization']
+                    .match(/\s.+$/)[0]
+                    .replace(/\s/, ''));
+            } catch(e) {
+                throw new Error('Could not parse authorization header.');
+            }
+
+
+            userinfo = jsonwebtoken.verify(auth_token, secret);
+
+            if(userinfo.aud !== (auth.client_id || DEFAULT_AUTH0_CLIENT_ID)) throw new Error('invalid \'aud\' claim');
+
+            if(auth.email) {
+                if(userinfo.email_verified === false) {
+                    throw new Error('This webtask can only be accessed if your ' + get_provider(userinfo.sub)  + ' email is verified');
+                }
+
+                if(auth.email instanceof Array) {
+                    for(var i = 0; i < auth.email.length; i++)
+                        if(userinfo.email.match(auth.email[i])) {
```

Comment:
It does work with domains at the moment because it just matches the given string against the email. Not sure on possible options to make this more bulletproof/harder to screw up from userland (could cast all the strings to `RegExp`s and append `$`?)